### ISSUE TITLE
The banner interferes with Emacs rainbow-delimiters and paredit

### ISSUE THREAD

**Issue by bigos** [lfe/lfe#312]
It's related to closed issue #215 
I have uploaded a video showing the problem.

https://www.youtube.com/watch?v=67diLo_gwZs

After closer investigation it looks like the | (vertical line) causes the problem.

`  
    (      \\     )    ;; |   A Lisp-2+ on the Erlang VM
`

Commenting out the vertical line character and the rest of the line seems to fix the problem.

**Comment by bigos** [comment]
Replacing | with ! also seems to fix the problem but introduces other oddities.
`  (      \\     )    !   A Lisp-2+ on the Erlang VM`
Emacs thinks the the line starts to the right from the exclamation mark. I do not need to edit the banner, so I can live with it, however it indicates that the problem goes deeper.

**Comment by yurrriq** [comment]
Thanks for the report. I thought we'd fixed this. I'll take a closer look soon, if no one else beats me to it.

**Comment by bigos** [comment]
One of possible fixes.
https://github.com/bigos/lfe/commit/3976dc45c7653c378381f0a82eb632a7ba05b2a6

In  https://github.com/rvirding/lfe/issues/215#issuecomment-204060675 yurrriq has suggested fixing the Emacs start-up file. I have tried it, but it didn't work. Perhaps, it's because my .emacs.d/ differs much from his start-up folder.

**Comment by bigos** [comment]
I did more experimenting with my Emacs start-up file. It looks like the code which is ran on 
`(add-hook 'inferior-lfe-mode-hook ...)`
has no effect on the content of *inferior-lfe* buffer. I tried (inferior-lfe-clear-buffer) and my own concoction. I experimented with inserting (debug) in the code and further evaluation shown that (buffer-string) of *inferior-lfe* buffer did not change. Strangely, once the REPL has appeared the code would work. 

The link below shows minimal fix of the problem. Replacing the first vertical line with exclamation mark fixes the problem.
https://github.com/bigos/lfe/commit/25c7519e4e066b1a86079a0632c1396b2b197236
Now the number of the vertical lines is even.

I would like to find a solution where REPL would be usable once it shows up without the need to do anything after that. I guess I could add another short-cut for the REPL clean-up. But I find it annoying.

On the other hand

**Comment by bigos** [comment]
I have experimented a bit with the issue and still have no solution. However Elisp fragment which can be evaluated in the *scratch* buffer can give someone some clues.

```
;;; Evaluating this progn will start lfe, but the REPL cleanup will not work.
(progn
  (run-lfe nil)
  (with-current-buffer "*inferior-lfe*"
    (inferior-lfe-clear-buffer)
    ;; print debug information
    (princ (format "going to print current buffer content %c" 10))
    (princ (buffer-string))
    (princ (format "%c above you can see current buffer content %c" 10 10))
    nil))

;;; Please note that the cleanup is not reflected in the REPL


;;; after starting lfe REPL evaluating this code will work
(with-current-buffer "*inferior-lfe*"
  (inferior-lfe-clear-buffer))

```

**Comment by bigos** [comment]
Patching make_banner() in such way that the last "|" is replaced with "!" also solves the problem. So perhaps number of the "|"s has to be even for all Emacs Lisp goodies to work? 

It's only one "|"character, which can be replaced with "!".

**Comment by bigos** [comment]
Adding following to my Emacs start-up fixes the problem as well, but in my opinion it is not a perfect solution. 

```
(global-set-key (kbd "s-q") 'inferior-lfe-clear-buffer)
```

**Comment by rvirding** [comment]
My guess is that it recognises `|` as a symbol quote character for `| making funny symbol names |` so it assumes they must come in pairs. And there is currently an odd number of `|` in the banner so removing one pairs them up.

Don't know why the standard lfe repl mode has no problems.

**Comment by yurrriq** [comment]
Did we ever get a no-banner option? I think that might solve it for @bigos too.

**Comment by rvirding** [comment]
No not yet. I have some problems running the "standard" inferior-lfe mode when you start to jump around the commands using the comint commands. Must try and find out why.

**Comment by oubiwann** [comment]
With the support of a `-nobanner` CLI arg for LFE, a possible fix has been created for this in #369. Note that it's currently blocked on another branch getting reviewed and merged.

**Comment by oubiwann** [comment]
`-nobanner` has been merged now; closing.
